### PR TITLE
[WIP] feat: deployment run

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #4321 #4340

### CODE REVIEW

File: packages/cli/src/abstract.command.ts [hunk 1]
```diff
@@ -104,10 +104,10 @@ export abstract class AbstractCommand {
...
 
     console.log(`
-      ${chalk.bold.white(config.name)}\t${config.description}
-      $ ${chalk.green('reapit')} ${parent ? `${chalk.whiteBright(parent.commandOptions.name)} ` : ''}${chalk.white(
-      config.name,
-    )} ${
+    ${parent ? '\t' : ''}${chalk.bold.white(config.name)}\t${config.description}
+    ${parent ? '\t' : ''}$ ${chalk.green('reapit')} ${
+      parent ? `${chalk.whiteBright(parent.commandOptions.name)} ` : ''
+    }${chalk.white(config.name)} ${
```
Comment (bashleigh) [inline packages/cli/src/abstract.command.ts:110]: tabbing formatting to produce a title with the parent command name and children tabbed in

File: packages/cli/src/abstract.command.ts [hunk 2]
```diff
@@ -104,10 +104,10 @@ export abstract class AbstractCommand {
     const args: ArgsType[] | undefined = Reflect.getOwnMetadata(ARGUMENT_OPTIONS, this.constructor)
 
     console.log(`
-      ${chalk.bold.white(config.name)}\t${config.description}
-      $ ${chalk.green('reapit')} ${parent ? `${chalk.whiteBright(parent.commandOptions.name)} ` : ''}${chalk.white(
-      config.name,
-    )} ${
+    ${parent ? '\t' : ''}${chalk.bold.white(config.name)}\t${config.description}
```
Comment (willmcvay) [inline packages/cli/src/abstract.command.ts:107]: This makes my head hurt a bit! Can we do anything to tidy up? Not a big deal though...

File: packages/cli/src/commands/deploy/list.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  async run() {
+    const spinner = ora()
+
+    spinner.start('fetching package info')
+    const packageInfo = await this.projectInfo()
+    spinner.succeed('found package info')
+
+    const deploys = await this.listDeployments(spinner, packageInfo.name)
+
+    console.table(
```
Comment (bashleigh) [inline packages/cli/src/commands/deploy/list.ts:45]: need to find a better alternative to this

File: packages/cli/src/commands/react-starter.ts [hunk 4]
```diff
@@ -54,6 +56,131 @@ export class ReactStarterCommand extends AbstractCommand {
     await this.writeConfigFile(`${path}/src/reapit.config.json`, reapitConfig)
   }
 
+  protected async createServerlessConfig(path: string): Promise<void> {
+    const content = `
```
Comment (willmcvay) [inline packages/cli/src/commands/react-starter.ts:60]: Might be nice to use a templating engine for this stuff - ejs, hbs or whatever

File: packages/cli/src/commands/help.ts [hunk 5]
```diff
@@ -4,11 +4,15 @@ import { ParentCommand } from './../parent.command'
 export class HelpCommand extends AbstractCommand {
   commands: (AbstractCommand | ParentCommand)[] = []
 
+  sortCommands(a: ParentCommand | AbstractCommand): number {
+    return a instanceof ParentCommand ? 1 : -1
+  }
```
Comment (bashleigh) [inline packages/cli/src/commands/help.ts:9]: eslint hated the shorthand sort func `sort((a, b) => a instanceof ParentCommand ? 1 : -1)` don't know why, something about the ternary return 🤷

**Review by willmcvay — APPROVED** [review]
Basically looks good as far as I can follow, nice one 👍🏻

**Pull request merged**